### PR TITLE
Clean up front page shelves experiment - keep saved searches, remove favorites

### PR DESCRIPTION
**Pull request by ninarg**
# PRs merged into this one
- #1181 

# Why?

We implemented an experiment showing your saved searches and recent favorites on the front page.
The experiment was actually never run for iOS for different reasons - but we will conclude based on Android's findings.

The decision was to:
- Remove favorite shelf
- Show saved searches for everyone (it opens our stories feature)

# What?

- Remove everything related to favorite shelf. 
- Simplify shelves into a single `FrontPageSavedSearchesView`
- Demo `FrontPageSavedSearchesView` and `FrontPageView`, and update snapshots.
- Delete demos for favorites and `FrontPageShelfDemoView`

# Version Change

Breaking change. 

# UI Changes

| Before | After |
| --- | --- |
|   |  |

### CODE REVIEW

File: FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift [hunk 1]
```diff
@@ -0,0 +1,183 @@
+import UIKit
+
+public protocol FrontPageSavedSearchesViewDelegate: AnyObject {
+    func frontPageSavedSearchesView(_ view: FrontPageSavedSearchesView, didSelectSavedSearch savedSearch: FrontPageSavedSearchViewModel)
+    func frontPageSavedSearchesViewDidSelectActionButton(_ view: FrontPageSavedSearchesView)
+}
+
+public class FrontPageSavedSearchesView: UIView {
```
Comment (ninarg) [inline FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift:8]: FYI - This file isn't new. It's renamed from `FrontPageShelfView` - with modifications to remove favorites.

File: FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Lifecycle
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        if let indexPath = scrollToIndexPath {
+            self.scrollToIndexPath = nil
+            // Using centeredHorizontally for the first cell to preserve leading content inset for iOS 14 and below.
```
Comment (MarenZR) [inline FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift:83]: Good use of comment 😊

File: FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+            let scrollPosition: UICollectionView.ScrollPosition = indexPath.item == 0 ? .centeredHorizontally : .left
+            collectionView.scrollToItem(at: indexPath, at: scrollPosition, animated: false)
+        }
+    }
+}
+
+// MARK: - Public methods
+
+public extension FrontPageSavedSearchesView {
+    func configure(with savedSearches: [FrontPageSavedSearchViewModel]) {
```
Comment (MarenZR) [inline FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift:93]: I'm just really curious as to why these extensions aren't just part of the class. It's consistent, so probably a reason for it?

**Review by MarenZR — APPROVED** [review]
Just some questions and comments, but nothing major. Love a good cleanup 🧹 God job! ✨

File: Demo/Sources/Components/FrontPageSavedSearchesDemoView/FrontPageSavedSearchesDemoView.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - Setup
+
+private extension FrontPageSavedSearchesDemoView {
```
[the remark this answers is not in the document]
Comment (ninarg) [reply Demo/Sources/Components/FrontPageSavedSearchesDemoView/FrontPageSavedSearchesDemoView.swift:26]: Good question. This file wasn't originally written by me, it's only renamed in this PR, while keeping the structure of the original file (GitHub makes everything look like it's new when renaming.. 😛). But I can change this, this file is small anyways so don't really need this separation

File: Demo/Sources/Components/FrontPageSavedSearchesDemoView/SavedSearchFactory.swift [hunk 5]
```diff
@@ -2,13 +2,13 @@ import Foundation
 import FinniversKit
 
 class SavedSearchShelfFactory {
```
[the remark this answers is not in the document]
Comment (ninarg) [reply Demo/Sources/Components/FrontPageSavedSearchesDemoView/SavedSearchFactory.swift:4]: Oh yes, nice find, renamed it ✅

File: FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift (hunk 3 above)
Comment (ninarg) [reply FinniversKit/Sources/Components/FrontPageSavedSearchesView/FrontPageSavedSearchesView.swift:93]: yes, as mentioned earlier it was just the coding style of the original developer of the files. Just another preference, I guess 😄  it's nice for separating concerns, so kept it as it was

File: FinniversKit/Sources/Fullscreen/FrontPage/FrontPageView.swift [hunk 6]
```diff
@@ -311,38 +312,43 @@ public final class FrontPageView: UIView {
...
+        self.savedSearchesViewModel = savedSearchesViewModel
+
+        if frontPageSavedSearchView == nil {
+            frontPageSavedSearchView = FrontPageSavedSearchesView(
+                title: savedSearchesViewModel.title,
+                buttonTitle: savedSearchesViewModel.buttonTitle,
+                remoteImageDataSource: remoteImageViewDataSource,
+                withAutoLayout: true
             )
+            savedSearchesContainer.addSubview(frontPageSavedSearchView!)
```
[the remark this answers is not in the document]
Comment (ninarg) [reply FinniversKit/Sources/Fullscreen/FrontPage/FrontPageView.swift]: done ✅

**Pull request merged**